cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/CombinedHBaseMetricsTable.java

```diff
@@ -151,20 +152,22 @@ public void delete(byte[] row, byte[][] columns) {
    * @param e
    */
   private void handleV2TableException(Exception e) {
-    if (e instanceof IOException) {
-      try {
-        if (!v2MetricsTableExists()) {
-          Closeables.closeQuietly(v2HBaseTable);
-          v2HBaseTable = null;
-          return;
+    if (e instanceof DataSetException || e instanceof RuntimeException) {
+      if (e.getCause() != null && e.getCause() instanceof IOException) {
+        try {
+          if (!v2MetricsTableExists()) {
+            Closeables.closeQuietly(v2HBaseTable);
+            v2HBaseTable = null;
+            return;
+          }
+        } catch (DatasetManagementException dme) {
+          LOG.error("Exception while checking dataset exists", dme);
```

Comment:
this is used for various purposes, not only existence check. You may need to pass in the operation to be able to issue an accurate message here.